vendor/bat-native-ads/src/bat/ads/internal/notifications.cc

```diff
@@ -49,76 +50,91 @@ void Notifications::Initialize(InitializeCallback callback) {
 bool Notifications::Get(const std::string& id, NotificationInfo* info) const {
   DCHECK(is_initialized_);
 
-  auto notification = notifications_.find(id);
-  if (notification == notifications_.end()) {
+  auto iter = std::find_if(notifications_.begin(), notifications_.end(),
+      [&id](const auto& notification){return notification.id == id;});
+  if (iter == notifications_.end()) {
     return false;
   }
 
-  *info = notification->second;
+  *info = *iter;
 
   return true;
 }
 
-void Notifications::Add(const NotificationInfo& info) {
+void Notifications::PushBack(const NotificationInfo& info) {
   DCHECK(is_initialized_);
 
-  notifications_.insert({info.id, info});
+  notifications_.push_back(info);
+  ads_client_->ShowNotification(std::make_unique<NotificationInfo>(info));
```

Comment:
@gdregalo Please revert this change as this function should not be responsible for showing the notification. @SergeyZhukovsky This is part of previous change which would be great if could be reverted before merging.

Reply:
Resolved in
https://github.com/brave/brave-core/pull/3499#discussion_r327762187